src/flamenco/vm/syscall/fd_vm_syscall_util.c

```diff
@@ -626,7 +668,9 @@ fd_vm_syscall_sol_memset( /**/            void *  _vm,
     fd_memset( dst, b, sz );
   } else {
     /* Syscall manages the pointer accesses directly and will report in the 
-       case of bad memory accesses. */
+       case of bad memory accesses. This syscall doesn't have the same nuance
+       as what is described in memcmp, because the VM will abort as soon as
+       out of bounds memory tries to get written to. */
```

Comment:
Therefore we don't have to be as careful about faulting early (with side effects) as we do in memcmp.